src/__tests__/wait-for-element-to-be-removed.js

```diff
@@ -0,0 +1,278 @@
+import {waitForElementToBeRemoved, wait} from '../'
+// adds special assertions like toBeTruthy
+import 'jest-dom/extend-expect'
+import {render} from './helpers/test-utils'
+import document from './helpers/document'
+
+const skipSomeTime = delayMs =>
+  new Promise(resolve => setTimeout(resolve, delayMs))
+
+// Using `setTimeout` >30ms instead of `wait` here because `mutationobserver-shim` uses `setTimeout` ~30ms.
+const skipSomeTimeForMutationObserver = (delayMs = 50) =>
+  skipSomeTime(delayMs, 50)
```

Comment:
Do you think we could use jest fake timers for this? I've found that tests that rely on setTimeout are pretty unreliable in CI environments especially.

Reply:
Sounds good, I will give a shot.

Reply:
I tried to use `fakeTimer`, but it does not seem to be an ideal place to use timeFakers.

As per I know, `setTimeout` are unreliable because of time-outs in CI right? How important do you think it is for this PR. If it's pretty important, I will try to look at it in more detail.

Reply:
I just don't want tests failing randomly on CI. And that happens when relying on setTimeout quite a bit.

Reply:
I have changed the tests. Can you please take a look.